### PR TITLE
MINOR: Enable capture of full stack trace in StreamTask#process

### PR DESCRIPTION
**Pull request by bbejeck**
Right now in `StreamTask#process` when we catch a `KafkaException`, only the message is printed in the log statement which makes debugging errors difficult.  This PR will capture the full stack trace of the exception and add it to the log statement.

I ran the existing streams test suite.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
\cc @ableegoldman  and @vvcephei for reviews as well

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java [hunk 1]
```diff
@@ -377,6 +383,18 @@ public boolean process() {
         return true;
     }
 
+    private String getStacktraceString(final KafkaException e) {
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java:396]: Maybe we could inline this, but my personal preference is to have a separate method call for readability.

**Review by guozhangwang — APPROVED** [review]
LGTM. call for second review @mjsax

**Comment by mjsax** [comment]
@bbejeck Can you rebase this to resolve the merge conflict?

**Comment by bbejeck** [comment]
rebased from trunk

**Comment by mjsax** [comment]
Unrelated test failures: https://builds.apache.org/job/kafka-pr-jdk8-scala2.11/19752/
```
java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:347)
	at scala.None$.get(Option.scala:345)
	at kafka.admin.PreferredReplicaLeaderElectionCommandTest.getLeader(PreferredReplicaLeaderElectionCommandTest.scala:91)
	at kafka.admin.PreferredReplicaLeaderElectionCommandTest.testNoopElection(PreferredReplicaLeaderElectionCommandTest.scala:230)
```
and
```
java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.CoordinatorNotAvailableException: The coordinator is not available.
	at org.apache.kafka.common.internals.KafkaFutureImpl.wrapAndThrow(KafkaFutureImpl.java:45)
	at org.apache.kafka.common.internals.KafkaFutureImpl.access$000(KafkaFutureImpl.java:32)
	at org.apache.kafka.common.internals.KafkaFutureImpl$SingleWaiter.await(KafkaFutureImpl.java:89)
	at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:260)
	at kafka.admin.ConsumerGroupCommand$ConsumerGroupService.resetOffsets(ConsumerGroupCommand.scala:306)
	at kafka.admin.ResetConsumerGroupOffsetTest.testResetOffsetsNotExistingGroup(ResetConsumerGroupOffsetTest.scala:89)
```

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.2`

**Pull request merged**